yggdrasilffi/src/lib.rs

```diff
@@ -123,8 +139,9 @@ fn result_to_json_ptr<T: Serialize>(result: Result<Option<T>, FFIError>) -> *mut
 /// `free_engine` and passing in the pointer returned by this method. Failure to do so will result in a leak.
 #[no_mangle]
 pub extern "C" fn new_engine() -> *mut c_void {
-    let engine = EngineState::default();
-    Box::into_raw(Box::new(engine)) as *mut c_void
+    let engine = Mutex::new(EngineState::default());
```

Comment:
This function is a the salient change. Rather than just a heap value, it's now a heap value wrapped in a thread safe read lock, which is wrapped in a reference counting pointer